index.js

```diff
@@ -58,8 +77,8 @@ function fastifyCors (fastify, opts, next) {
         )
       }
 
-      if (req.raw.method === 'OPTIONS' && preflight === true) {
-        // preflight
+      // Handle preflight headers (if strict mode is enabled, then the valid preflight headers must exit)
+      if (req.raw.method === 'OPTIONS' && preflight === true && (strictPreflight === false || (req.headers.origin && req.headers['access-control-request-method']))) {
```

Comment:
I think we should move this if inside the route.

Reply:
I started that way, but then decided to leave the headers stuff in the hook to keep the behavior similar to the previous `preflightContinue` behavior. The previous behavior would still add the headers, but leave the application to reply.
:point_up_2: 
I guess I took the perspective of minimal functional difference, but I see your point that it is just not the "fastify" way to do plugins and probably makes it more complicated to understand.

I'll see what it looks like if I move this header handling into the route.